kuzzleio/documentation-V2
# Add new scaffolding tool

## What does this PR do?

This PR add a scaffolding tool for the new documentation hierarchy.  

The scaffolding tool use templates to create files needed to document a new controller action. It uses the subsection format we agreed to.

There is two command: 
 - `generate <path>`: Render raw templates for a controller action
 - `copy <src> <dest>`: Render raw templates for `src` and extract shareable informations from `dest` to inject it in `src` templates (action description, arguments table, arguments description, test config file)

After generating the files, the script will extract the method signature from the SDK and the action description from the old documentation so you can put it in your document.

### How should this be manually tested?

  - Step 1: `./scaffolding/scaffold generate src/sdk-reference/js/6/admin/resetCache`
  - Step 2: Should create the following file in the directory: `index.md`, `snippets/resetCache.test.yml`, `snippets/resetCache.js`

  - Step 1: `./scaffolding/scaffold copy src/sdk-reference/js/6/admin/resetCache src/sdk-reference/cpp/1/admin/resetCache`
  - Step 2: Should create the following file in the directory with informations extracted from sdk js 6: `index.md`, `snippets/resetCache.test.yml`, `snippets/resetCache.js`


## Other changes

 - Add a DEV_MODE variable which avoid the download of existing SDK when running snippet tests. 
Usage: 
```bash
DEV_MODE=true bash run-snippet-tests.sh -n -p src/sdk-reference/java/1/index/create
# or
export DEV_MODE=true
```
